src/runtime.rs

```diff
@@ -501,6 +545,10 @@ impl<DB: Database> ActiveQuery<DB> {
         self.changed_at.is_constant = false;
         self.changed_at.revision = changed_at;
     }
+
+    fn add_anon_read(&mut self, changed_at: Revision) {
+        self.changed_at.revision = changed_at;
```

Comment:
Why this is not a `self.changed_at.revision = self.changed_at.revision.max(changed_at);`?

This seems like a bug, if a query checks for cancellation not on entry, but in the middle.

Reply:
um yeah that is wrong

Reply:
copy and paste error, good catch

Reply:
I should make a test that exposes it :)

Reply:
actually, I think I already did, I was wondering why this local test I have wasn't passing

Reply:
Ah, no, the code is wrong, but I don't think it's **observably** wrong. This is because `next_uncanceled_revision`, in fact, is always equal to either the current revision or the next one -- it is updated on every revision bump, since we don't know whether anyone is listening or not. I don't actually think this is harmful, although it leads to higher "changed at" numbers (or would, without "backdating").

I'll maybe rename it to `pending_revision` though